packages/shared/__tests__/ReactError-test.internal.js

```diff
@@ -39,20 +41,35 @@ describe('ReactError', () => {
   // @gate build === "production"
   // @gate !source
   it('should error with minified error code', () => {
-    expect(() => ReactDOM.render('Hi', null)).toThrowError(
+    expect(() => {
+      ReactDOMClient.createRoot(null);
+    }).toThrowError(
       'Minified React error #200; visit ' +
         'https://reactjs.org/docs/error-decoder.html?invariant=200' +
         ' for the full message or use the non-minified dev environment' +
         ' for full errors and additional helpful warnings.',
     );
   });
 
-  it('should serialize arguments', () => {
+  // @gate build === "production"
+  // @gate !source
+  it('should serialize arguments', async () => {
     function Oops() {
-      return;
+      return {};
     }
     Oops.displayName = '#wtf';
+
     const container = document.createElement('div');
-    expect(() => ReactDOM.render(<Oops />, container)).not.toThrowError();
+    const root = ReactDOMClient.createRoot(container);
+    await expect(async () => {
+      await act(async () => {
+        root.render(<Oops />);
+      });
+    }).rejects.toThrow(
+      'Minified React error #152; visit ' +
+        'https://reactjs.org/docs/error-decoder.html?invariant=152&args[]=%23wtf' +
+        ' for the full message or use the non-minified dev environment' +
+        ' for full errors and additional helpful warnings.',
+    );
```

Comment:
Reverts this back to what it was before https://github.com/facebook/react/pull/21869

In that PR, we allow returning undefined from render. The test was updated (stupidly, by me) to just pass. But the test is to confirm arguments are serialized correctly, so the right fix is to change it to an error that includes arguments. 

I've chosen rendering an object as the error.